math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -848,6 +848,59 @@ RVec<RVec<typename RVec<T>::size_type>> Combinations(const RVec<T>& v, const typ
    }
 }
 
+/// Return the indices of the elements which are not zero
+template <typename T>
+RVec<typename RVec<T>::size_type> Nonzero(const RVec<T> &v)
+{
+   using size_type = typename RVec<T>::size_type;
+   RVec<size_type> r;
+   for(size_type i=0; i<v.size(); i++) {
+      if(v[i] != 0) {
+         r.emplace_back(i);
+      }
+   }
+   return r;
+}
+
+/// Return the intersection of elements of two vectors.
+/// Each element of v1 is looked up in v2 and added to the returned vector if
+/// found. Following, the order of v1 is reserved. Repetitions of elements in
+/// v2 are added multiple times to the returned vector if allow_repetitions is
+/// set to true;
+template <typename T>
+RVec<T> Intersect(const RVec<T>& v1, const RVec<T>& v2, bool allow_repetitions = false)
+{
+   // TODO: Remove comment after review.
+   // NOTE: This can be done much more efficient (for large vectors) if we sort
+   // v1 and v2 before.
+   RVec<T> r;
+   using size_type = typename RVec<T>::size_type;
+   for(size_type i=0; i<v1.size(); i++) {
+      for(size_type j=0; j<v2.size(); j++) {
+         if(v1[i] == v2[j]) {
+            r.emplace_back(v1[i]);
+            if(!allow_repetitions) {
+                continue;
+            }
+         }
+      }
+   }
+   /*
+   TODO: Remove comment after review.
+
+   std::set_intersection(v1.begin(), v1.end(),
+                         v2.begin(), v2.end(),
+                         std::back_inserter(r));
+
+   This works not because of this example:
```

Comment:
Yes, of course, you're quite right!  It would need `v1` and `v2` to be ordered, my mistake!

Reply:
It took me some time to figure out the STL algorithm actually does :D